input_output.ipynb

```diff
@@ -0,0 +1,919 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Writing to a file\n",
+    "The process to write data to a file is very similar, the main difference being that:\n",
+    "- We use `w` as a second argument of `open` to specify that we want to write to the file. If the file already exists, it will be erased before we write something else to it. If you want to append to the file, you should use `a` instead.\n",
+    "- We use `write` to write a *string* to the file. Other types of object should be converted to string before being written.\n",
+    "\n",
+    "\n",
+    "Let's see this in action by writing your name in a file called `me.txt` in [data](./data/)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pathlib as pl\n",
+    "\n",
+    "path = pl.Path(\"./tutorial/tests/data/me.txt\")\n",
+    "file_ob = open(path, \"w\")\n",
+    "file_ob.write(\"Simone\")\n",
+    "file_ob.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Congratulations! Your name is now written in stone. \n",
+    "\n",
+    "If we want to write the contents of an *iterable* to a file, we can use the `writelines` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pathlib as pl\n",
+    "\n",
+    "path = pl.Path(\"./tutorial/tests/data/numbers.txt\")\n",
+    "file_ob = open(path, \"w\")\n",
+    "file_ob.writelines([str(i) + \"\\n\" for i in range(10)])\n",
+    "file_ob.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that for each line, we concatenate the `newline` `\\n` symbol to the string to be written to write the text to a new line.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "### Context managers\n",
+    "As you can see, after *opening* and performing operations on a file, we always have to remember to *close* it. If we forget to close it, unexpected behavior can happen. If the program crashes later on, for example, we might have the situation where the text is not written to the file. If you open many files and you don't close them, the python intepreter can run out of memory. On some operating systems, the file contents are only updated after closing, etc ...\n",
+    "\n",
+    "This pattern is very common dealing with many *resources*: files, connections, threads, servers, etc...  You acquire the access to the resource, do some work on it and finally you clean up after yourself by closing it again. Because of this, python offers a construct called [*context manager*](https://docs.python.org/3/reference/datamodel.html#context-managers) which implements exactly this beahvior:\n",
```

Comment:
```suggestion
    "This pattern is common when dealing with many *resources*: files, connections, threads, servers, etc...  You acquire access to the resource, do some work on it, and, finally, clean up after yourself by closing it again. Because of this, python offers a construct called [*context manager*](https://docs.python.org/3/reference/datamodel.html#context-managers) which implements exactly this beahvior:\n",
```